src/emulation/mod.rs

```diff
@@ -78,14 +78,12 @@ impl SyscallMock {
             }
             (Syscall::Stat, SyscallStop::Exit) if !self.protocol.mocked_files.is_empty() => {
                 let filename = tracee_memory::peek_string(pid, registers.rdi)?;
-                println!(
-                    "mocked files: {:?} - {:?}",
-                    self.protocol.mocked_files.clone().into_iter().map(|f| String::from_utf8(f).unwrap()).collect::<Vec<String>>(),
-                    String::from_utf8(filename.clone())
-                );
                 if self.protocol.mocked_files.contains(&filename) {
-                    println!("huhu");
-                    let mut registers = registers.clone();
+                    if filename.ends_with(b"/") {
+                        let buffer_ptr = registers.rsi;
+                        tracee_memory::pokedata(pid, buffer_ptr + 24, u64::from(libc::S_IFDIR))?;
```

Comment:
@soenkehahn i figured out how to find the location in memory where the file mode is specified in `*statbuf` 😄 (this was a stretch for me... some relevant links [statbuf struct def](https://doc.servo.org/libc/unix/notbsd/linux/other/b64/x86_64/struct.stat.html), [descriptions of the fields](http://man7.org/linux/man-pages/man2/fstat.2.html), [extended info about the fields](http://man7.org/linux/man-pages/man7/inode.7.html))

...but i'm going to need some advice on how to clean this up... i'm not in love with the magic number, but unwrapping the entire struct isn't really necessary. also `pokedata` and `ptrace::write` write 8 bytes but `S_IFDIR` is actually a `u32` (so i assume that means i'm clobbering the `st_uid` field, which could have side effects).